src/simplifiers/BoolRewriting.cpp

```diff
@@ -112,8 +112,10 @@ PTRef mergePTRefArgs(Logic & logic, PTRef tr, Map<PTRef,PTRef,PTRefHash>& cache,
     Pterm& t = logic.getPterm(tr);
     SymRef sr = t.symb();
     vec<PTRef> new_args;
+    bool changed = false;
     for (int i = 0; i < t.size(); i++) {
         PTRef subst = cache[t[i]];
+        changed |= subst != t[i];
```

Comment:
Maybe add the parentheses here:
changed |= (subst != t[i]);

Reply:
Fixed in 4cded77